Mathlib/Analysis/Meromorphic/TrailingCoefficient.lean

```diff
@@ -0,0 +1,258 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.Analysis.Meromorphic.Order
+
+/-!
+# The Trailing Coefficient of a Meromorphic Function
+
+This file defines the trailing coefficient of a meromorphic function. If `f` is meromorphic at a
+point `x`, the trailing coefficient is defined as the (unique!) value `g x` for a presentation of
+`f` in the form `(z - x) ^ order • g z` with `g` analytic at `x`.
+
+The lemma `meromorphicTrailingCoeffAt_eq_limit` expresses the trailing coefficient as a limit.
+-/
+
+variable
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {f g : 𝕜 → E} {x : 𝕜} {n : ℤ}
+
+open Filter Topology
+
+variable (f x) in
+/--
+If `f` is meromorphic of finite order at a point `x`, the trailing coefficient is defined as the
+(unique!) value `g x` for a presentation of `f` in the form `(z - x) ^ order • g z` with `g`
+analytic at `x`. In all other cases, the trailing coefficient is defined to be zero.
+-/
+noncomputable def meromorphicTrailingCoeffAt : E := by
+  by_cases h₁ : MeromorphicAt f x
+  · by_cases h₂ : meromorphicOrderAt f x = ⊤
+    · exact 0
+    · exact ((meromorphicOrderAt_ne_top_iff h₁).1 h₂).choose x
+  · exact 0
+
+/--
+If `f` is not meromorphic at `x`, the trailing coefficient is zero by definition.
+-/
+@[simp] lemma meromorphicTrailingCoeffAt_of_not_MeromorphicAt (h : ¬MeromorphicAt f x) :
+    meromorphicTrailingCoeffAt f x = 0 := by simp_all [meromorphicTrailingCoeffAt]
```

Comment:
```suggestion
    meromorphicTrailingCoeffAt f x = 0 := by simp [meromorphicTrailingCoeffAt, h]
```
In general, `simp_all` is not really advisable because it is less predictable than a bare `simp`.